### PR TITLE
Cleanup sidebar CSS

### PR DESCRIPTION
**Pull request by robertknight**
This PR cleans up the CSS for the top-level `HypothesisApp` component and the entrypoint `sidebar.scss` file:

- CSS classes for `HypothesisApp` were renamed to `hypothesis-app*` and moved to `src/styles/sidebar/components/hypothesis-app.scss` to follow our usual conventions
- Unused CSS classes from `sidebar.scss` and `common.scss` have been removed
- The `sheet` styles from `sidebar.scss` have been consolidated into `focused-mode-header`, since the only element that used them also had that style
- Unused SASS module imports have been removed from `sidebar.scss`
- Element styles from `common.scss` have been consolidated in `elements.scss` (for non-`body` styles) or the bottom of `sidebar.scss` (for `body` styles)
- The `content` CSS class that was used by `TopBar` and `HypothesisApp` to center the content of the sidebar/stream when the page is wide enough has been extracted into a mixin (`sidebar-content` in `mixins/layout.scss`) which is now used by the respective component styles

### CODE REVIEW

File: src/sidebar/components/focused-mode-header.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default function FocusedModeHeader() {
   })();
 
   return (
-    <div className="focused-mode-header sheet sheet--short">
+    <div className="focused-mode-header">
```
Comment (robertknight) [inline src/sidebar/components/focused-mode-header.js:52]: The `sheet` styles have been consolidated into the `focused-mode-header` class.

File: src/styles/sidebar/components/focused-mode-header.scss [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
   align-items: center;
+  position: relative;
+
+  background-color: var.$body-background;
+  border-radius: 2px;
+  border: solid 1px var.$grey-3;
+  font-family: var.$sans-font-family;
+  font-weight: 300;
+  margin-bottom: 0.72em;
+  padding: 0.5em 1em;
```
Comment (robertknight) [inline src/styles/sidebar/components/focused-mode-header.scss:18]: These come from the `sheet` class which used to live in `sidebar.scss`

File: src/styles/sidebar/elements.scss [hunk 3]
```diff
@@ -16,6 +16,18 @@ a:hover {
...
+  font-style: italic;
+}
+
+p {
+  hyphens: auto;
+
+  & + p {
+    margin: 1em 0 0;
+  }
+}
```
Comment (robertknight) [inline src/styles/sidebar/elements.scss:29]: These were moved from `common.scss`

File: src/styles/sidebar/sidebar.scss [hunk 4]
```diff
@@ -72,106 +61,14 @@
...
-
-.sheet--is-theme-clean {
-  padding-left: 30px;
-  padding-bottom: 30px;
-  margin-bottom: 20px;
-  margin-left: 5px;
-  margin-right: 5px;
-}
-
-.annotation-unavailable-message {
```
Comment (robertknight) [inline src/styles/sidebar/sidebar.scss:158]: This is unused

File: src/styles/sidebar/sidebar.scss [hunk 5]
```diff
@@ -72,106 +61,14 @@
...
-// Disable scroll anchoring as it interferes with `ThreadList` and
-// `visible-threads` calculations and can cause a render loop
-.js-thread-list-scroll-root {
-  overflow-anchor: none;
-}
-
-// Elements in root template (viewer.html)
-// ---------------------------------------
 
-.sheet {
```
Comment (robertknight) [inline src/styles/sidebar/sidebar.scss:129]: The used `sheet*` styles have been moved to `focused-mode-header`

File: src/styles/sidebar/sidebar.scss [hunk 6]
```diff
@@ -72,106 +61,14 @@
...
-    background-image: url(../images/icons/lock.svg);
-    background-repeat: no-repeat;
-    width: 56px;
-    height: 48px;
-  }
+  color: var.$text-color;
+  font-family: var.$sans-font-family;
+  font-size: var.$base-font-size;
+  font-weight: normal;
+  line-height: var.$base-line-height;
```
Comment (robertknight) [inline src/styles/sidebar/sidebar.scss:73]: These are consolidated from styles that used to be in `common.scss`

File: src/styles/sidebar/components/hypothesis-app.scss [hunk 7]
```diff
@@ -0,0 +1,29 @@
+@use "../../variables" as var;
+@use "../mixins/layout";
+@use "../mixins/responsive";
+
+.hypothesis-app {
+  $sidebar-h-padding: 9px;
```
Comment (lyzadanger) [inline src/styles/sidebar/components/hypothesis-app.scss:6]: 9 pixels, eh? :)

Comment (robertknight) [reply src/styles/sidebar/components/hypothesis-app.scss:6]: 🤷

**Pull request merged**